pymc3/sampling.py

```diff
@@ -134,6 +151,20 @@ def sample(draws, step=None, init='ADVI', n_init=200000, start=None,
         in the system - 2.
     tune : int
         Number of iterations to tune, if applicable (defaults to None)
+    step_kwargs: dict
```

Comment:
I understand you added the step-method name to also support auto-assignment. In the 95% case, however, people would just pass in nuts kwargs in which case `sample_kwargs={'target_accept': .9}` is more readable and direct than: `sample_kwargs={'nuts': {'target_accept': .9}}`. This current one might be the best solution but it's worthwhile to consider the alternatives:
* Raise exception when step_kwargs is used for non-continuous models (unsupported).
* Accept single-nested dict for continuous models (single step-method) but also double-nested for mixed and continuous.

The second one is a bit less clean but I think fine at the API level, and it would just be one more if. I think that option gets my vote. @fonnesbeck?